src/simple_model.jl

```diff
@@ -128,7 +128,7 @@ end
 
 function jprod!(nlp :: SimpleNLPModel, x :: Vector, v :: Vector, Jv :: Vector)
   nlp.counters.neval_jprod += 1
-  Jv[:] = ForwardDiff.jacobian(nlp.c, x) * v
+  Jv[1:nlp.meta.ncon] = ForwardDiff.jacobian(nlp.c, x) * v
```

Comment:
So now I'm wondering if this shouldn't remain `Jv[:]` and if the caller should use `sub(...)`. What do you think?